code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -50,94 +61,161 @@
+		H.update_inv_wear_suit()
+	for(var/datum/action/A in actions)
+		A.build_all_button_icons()
 	return
 
+/obj/item/clothing/suit/armor/reactive/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text, damage, attack_type)
+	. = ..()
+
+	if(!active || !prob(hit_reaction_chance))
+		return FALSE
+	if(reactivearmor_cooldown_duration && !COOLDOWN_FINISHED(src, reactivearmor_cooldown))
+		cooldown_activation(owner)
+		return FALSE
+	if(reactivearmor_cooldown_duration)
+		COOLDOWN_START(src, reactivearmor_cooldown, reactivearmor_cooldown_duration)
+
+	if(bad_effect)
+		return emp_activation(owner, hitby, attack_text, damage, attack_type)
+	else
+		return reactive_activation(owner, hitby, attack_text, damage, attack_type)
+
+/**
+ * A proc for doing cooldown effects (like the sparks on the tesla armor, or the semi-stealth on stealth armor)
+ * Called from the suit activating whilst on cooldown.
+ * You should be calling ..()
+ */
+/obj/item/clothing/suit/armor/reactive/proc/cooldown_activation(mob/living/carbon/human/owner)
+	owner.visible_message(cooldown_message)
+
+/**
+ * A proc for doing reactive armor effects.
+ * Called from the suit activating while off cooldown, with no emp.
+ * Returning TRUE will block the attack that triggered this
+ */
+/obj/item/clothing/suit/armor/reactive/proc/reactive_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>The reactive armor doesn't do much! No surprises here.</span>")
+	return TRUE
+
+/**
+ * A proc for doing owner unfriendly reactive armor effects.
+ * Called from the suit activating while off cooldown, while the armor is still suffering from the effect of an EMP.
+ * Returning TRUE will block the attack that triggered this
+ */
+/obj/item/clothing/suit/armor/reactive/proc/emp_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>The reactive armor doesn't do much, despite being emp'd! Besides giving off a special message, of course.</span>")
+	return TRUE
+
 /obj/item/clothing/suit/armor/reactive/emp_act(severity)
 	. = ..()
-	if(. & EMP_PROTECT_SELF)
+	if(. & EMP_PROTECT_SELF || bad_effect || !active) //didn't get hit or already emp'd, or off
 		return
-	active = 0
-	icon_state = "reactiveoff"
-	item_state = "reactiveoff"
-	reactivearmor_cooldown = world.time + 200
+	if(ismob(loc))
+		to_chat(loc, emp_message)
+	bad_effect = TRUE
+	addtimer(VARSET_CALLBACK(src, bad_effect, FALSE), 30 SECONDS)
 
 //When the wearer gets hit, this armor will teleport the user a short distance away (to safety or to more danger, no one knows. That's the fun of it!)
 /obj/item/clothing/suit/armor/reactive/teleport
 	name = "reactive teleport armor"
 	desc = "Someone separated our Research Director from his own head!"
 	var/tele_range = 6
 	var/rad_amount= 15
```

Comment:
newly defined vars should be at the bottom of the var list (under ``reactivearmor_cooldown_duration`` in this case)